tests/unit/test_signers.py

```diff
@@ -401,6 +469,38 @@ def test_generate_presigned_post(self):
                          'https://s3.amazonaws.com/mybucket')
         self.assertEqual(post_form_args['fields'], {})
 
+    def test_generate_presigned_post_emits_choose_signer(self):
+        with mock.patch.dict(botocore.auth.AUTH_TYPE_MAPS,
+                             {'s3v4-presign-post': self.auth}):
+            self.signer.generate_presigned_post(self.request_dict)
+        self.emitter.emit_until_response.assert_called_with(
+            'choose-signer.s3.PutObject',
+            signing_name='signing_name', region_name='region_name',
+            signature_version='s3v4-presign-post')
+
+    def test_generate_presigned_post_choose_signer_override(self):
+        auth = mock.Mock()
+        self.emitter.emit_until_response.return_value = (None, 'custom')
```

Comment:
Should we add test cases for when you return a ``s3v4-presign-post`` or ``s3-query``?